### PR TITLE
Player Spawning and Multiplayer Support

### PR DESCRIPTION
**Pull request by anjx123**
Players now randomly spawn around a fridge and you can cycle through multiple players using the T button

### CODE REVIEW

File: Assets/Code/Scripts/Player/Player.cs [hunk 1]
```diff
@@ -5,7 +5,7 @@
 public class Player : MonoBehaviour
 {
     [Header("Debug")]
-    public bool controlToggle;
+    public int controlToggle = 0;
```
Comment (Hanson-Sun) [inline Assets/Code/Scripts/Player/Player.cs:8]: cursor

File: Assets/Code/Scripts/Player/Player.cs [hunk 2]
```diff
@@ -16,6 +16,15 @@ public class Player : MonoBehaviour
 
     public List<Buff> buffs = new List<Buff>();
 
+    private int playerID;
+
+
+    public int PlayerID
+    {
+        get { return playerID; }
```
Comment (Hanson-Sun) [inline Assets/Code/Scripts/Player/Player.cs:24]: use the funny `=>`

File: Assets/Code/Scripts/Player/Player.cs (hunk 1 above)
Comment (anjx123) [reply Assets/Code/Scripts/Player/Player.cs:8]: What is Cursor

File: Assets/Code/Scripts/Player/Player.cs (hunk 2 above)
Comment (anjx123) [reply Assets/Code/Scripts/Player/Player.cs:24]: Wo Bu Zhi Dao Zen Me

File: Assets/Code/Scripts/Systems/SceneUtility/GameManager.cs [hunk 3]
```diff
@@ -6,9 +6,18 @@ public class GameManager : MonoBehaviour
...
     //public so they can be viewed in GUI
-    public List<Player> players;
-    public List<Projectile> projectiles;
-    public List<Trap> traps;
+    [SerializeField]
+    private List<Player> players;
+
+    public List<Player> Players
+    {
+        get { return players; }
```
Comment (Hanson-Sun) [inline Assets/Code/Scripts/Systems/SceneUtility/GameManager.cs:14]: `get => players`

File: Assets/Code/Scripts/Systems/SceneUtility/PlayerSpawn.cs [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        Vector3 spawnAreaCenter = spawnCenter.position + offset;
+        Vector3 spawnAreaSize = new Vector3(spawnAreaWidth, spawnAreaLength, 0f);
+        Vector3 spawnPoint = Vector3.zero;
+
+        bool validSpawnPointFound = false;
+        int maxAttempts = 10;
+
+        for (int j = 0; j < numOfPlayers; ++j)
+        {
+            for (int i = 0; i < maxAttempts; i++)
```
Comment (Hanson-Sun) [inline Assets/Code/Scripts/Systems/SceneUtility/PlayerSpawn.cs:40]: maxAttempts 😰, bro is preventing infinite while loops

**Review by Hanson-Sun — APPROVED** [review]
ok looks fine, i didnt run it since unity takes 2 years to start up

**Pull request merged**